### DISCUSSION TITLE
DetectEdges and MoothMeshLines

### DISCUSSION THREAD

**Discussion by bernienor** [thliebig/openEMS-Project#210] [Q&A Usage]
Hi!

I've made a simple example of a monopole antenna. It consists of a disc as the ground plane and a wire as the antenna. 

I'm not sure how to best mesh this example. The antenna wire is a wire defined from 0.5 aboved the ground plane and with a length "AntennaLength" along the Z-axis. The ground plane is in the XY plane. The WireRadius 0.25 mm

Using DetectEges gives me a warning:

```
CSX` = AddMetal( CSX, 'monopole' ); % create a perfect electric conductor (PEC)
CSX = AddWire(CSX, 'monopole', 10, points, WireRadius); %

% Meshing the wire:
antenna_mesh = DetectEdges(CSX, [], 'SetProperty','monopole');
mesh.z = [mesh.x SmoothMeshLines(antenna_mesh.z, 0.5)];
mesh.x = [mesh.x SmoothMeshLines(antenna_mesh.x, 0.5)];
mesh.y = [mesh.y SmoothMeshLines(antenna_mesh.y, `0.5)];
```

results in:

```
warning:` unsupported primitive of type: "Wire" found, skipping edges
warning: called from
    DetectEdges at line 214 column 25
    monopole at line 91 column `6
```

Would it be sufficient to add meshlines at Z= AntennaLength (and IE. +/- 0.5 mm)?
How would you recomend to best set up the mesh to get good simulation results?

Cheers!
Bernie

**Comment by georgy-moshkin** [comment]
Maybe something like this?
```
mesh.x = [mesh.x linspace(wire.centerX-WireRadius, wire.centerX+WireRadius, 10)];
mesh.z = [mesh.z linspace(wire.centerZ-WireRadius ,wire.centerZ+WireRadius, 10)];
```
or

```
for angle= 0:35
mesh.x = [mesh.x wire.centerX+(WireRadius+0.1)*cos(angle*10*3.14/180)];
mesh.z = [mesh.z wire.centerZ+(WireRadius+0.1)*sin(angle*10*3.14/180)];
mesh.x = [mesh.x wire.centerX+(WireRadius-0.1)*cos(angle*10*3.14/180)];
mesh.z = [mesh.z wire.centerZ+(WireRadius-0.1)*sin(angle*10*3.14/180)];
endfor
```

Only as idea, not tested

**Reply by VolkerMuehlhaus** [reply]
I am not familar with the openEMS wire model, not sure what the diameter parameter does there? Does that really add a thick wire, or use a thin wire plus some numerical compensation to approximate the thickness?

When I model wire antennas in FDTD, I use **square** cross section, and place mesh lines at the boundary of the square, but not inside. Trying to model and mesh the **round** cross section will slow down simulation by generating a smaller time step, so I would rather avoid that.
